bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+        boolean nextCharLow = true;
+
+        for (int i = 0; i < string.length(); i++) {
+            char currentChar = string.charAt(i);
+            if (delimiter == currentChar) {
+                nextCharLow = false;
+            } else if (nextCharLow) {
+                builder.append(Character.toLowerCase(currentChar));
+            } else {
+                builder.append(Character.toUpperCase(currentChar));
+                nextCharLow = true;
+            }
+        }
+        return builder.toString();
+    }
+
+    private @Nullable String sendGetRequest(String url, String... params) {
+        String errorReason = "";
+        if (config == null) {
+            errorReason = "Invalid Binding configuration";
+        } else if (config.hostname.isEmpty()) {
+            errorReason = "Invalid hostname configuration";
+        } else {
+            String req = "http://" + config.hostname + url;
+            if (!config.key.isEmpty()) {
+                req += "?key=" + config.key;
+            }
+            for (int i = 0; i < params.length; i++) {
+                req += "&" + params[i];
+            }
+
+            Request request = httpClient.newRequest(req);
+            request.method(HttpMethod.GET).timeout(5, TimeUnit.SECONDS).header(HttpHeader.ACCEPT_ENCODING, "gzip");
+
+            try {
+                ContentResponse contentResponse = request.send();
+                if (contentResponse.getStatus() == 200) {
+                    if (!this.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                        updateStatus(ThingStatus.ONLINE);
+                    }
+                    try {
+                        String response = new String(contentResponse.getContent(), Charset.forName(config.encoding));
+                        if (url == DAQEXTDESC_URL) {
+                            parseAndJsonInit(response);
+                        } else if (url == DAQDATA_URL) {
+                            parseAndUpdate(response);
+                        } else if (url == DAQDESC_URL) {
+                            parseAndInit(response);
+                        } else if (url == PARSET_URL) {
+                            // via return
+                        }
+                        return response;
+                    } catch (IllegalArgumentException e) {
+                        errorReason = String.format("IllegalArgumentException: %s", e.getMessage());
+                    }
+                } else {
+                    errorReason = String.format("Guntamatic request failed with %d: %s", contentResponse.getStatus(),
+                            contentResponse.getReason());
+                }
+            } catch (TimeoutException e) {
+                errorReason = "TimeoutException: Guntamatic was not reachable on your network";
+            } catch (ExecutionException e) {
+                errorReason = String.format("ExecutionException: %s", e.getMessage());
```

Comment:
Could lead to NPE if e.getMessage() is null

Reply:
`((e.getMessage() != null) ? e.getMessage() : "")`